core/src/main/scala/io/finch/RequestReaders.scala

```diff
@@ -135,6 +136,18 @@ trait RequestReaders {
   val binaryBody: RequestReader[Array[Byte]] = binaryBodyOption.failIfNone
 
   /**
+   * A [[RequestReader]] that reads a required chunked streaming binary body, interpreted as a `AsyncStream[Buf]`.
+   */
+  def asyncBody(chunkSize: Int): RequestReader[AsyncStream[Buf]] = {
```

Comment:
Looks great! I'm not sure we need to give users `chunkSize` to specify. We could (and I think) should use `Int.MaxValue` so we will read the entire chunk sent by a client. Let's make it `val asyncBody`.